### PR TITLE
Use version instead of short version

### LINKED ISSUE #1241: Use VERSION instead of SHORT_VERSION
**Component**:

'salt', 'build'

**Why this is needed**:

Currently we can't differenciate two minor version (ISO mounted as the same place, same saltenv, ...)

**What should be done**:

Use `VERSION` instead of `SHORT_VERSION` everywhere

**Implementation proposal** (strongly recommended):

Some WIP https://github.com/scality/metalk8s/commit/c2aa4c538a0f2b8f3912887873e05899375a85a2

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
**Component**:

'salt', 'kubernetes', 'build'

**Context**: 

Has we will likely upgrade from minor version we need to use `VERSION` instead of `SHORT_VERSION` everywhere

**Summary**:

Replace `SHORT_VERSION` by `VERSION` everywhere

**Acceptance criteria**: 

Have a working deployment using `metalk8s-<version>` as saltenv

---

Fixes: #1241

### CODE REVIEW

**Comment by TeddyAndrieux** [comment]
@bert-e reset

**Comment by TeddyAndrieux** [comment]
@bert-e reset

File: salt/_modules/metalk8s.py [hunk 1]
```diff
@@ -225,8 +225,15 @@ def get_products(products=None):
             continue
 
         info.update({'path': path, 'iso': iso})
-        res.update(
-            {'metalk8s-{0}'.format(version): info}
-        )
+        env_name = 'metalk8s-{0}'.format(version)
```
Comment (slaperche-scality) [inline salt/_modules/metalk8s.py:228]: What's the relation with this PR?

File: buildchain/buildchain/salt_tree.py [hunk 2]
```diff
@@ -277,7 +277,7 @@ def _run(self) -> None:
     Path('salt/metalk8s/repo/configured.sls'),
     Path('salt/metalk8s/repo/deployed.sls'),
     Path('salt/metalk8s/repo/files/nginx.conf.j2'),
-    Path('salt/metalk8s/repo/files/90-metalk8s-registry-config.inc.j2'),
```
Comment (slaperche-scality) [inline buildchain/buildchain/salt_tree.py:280]: Why this change? How is it related so `s/SHORT_VERSION/VERSION/`?

File: buildchain/buildchain/salt_tree.py [hunk 3]
```diff
@@ -357,17 +357,16 @@ def _run(self) -> None:
...
+            constants.VERSION.replace('.', '_').replace('-', '_')
         ),
         name_prefix='{}-{}/'.format(
-            config.PROJECT_NAME.lower(), constants.SHORT_VERSION
+            config.PROJECT_NAME.lower(), constants.VERSION
         ),
         destination=Path(
             constants.ISO_ROOT,
             'salt/metalk8s/repo/files',
-            '99-{}-{}-registry.inc'.format(
```
Comment (slaperche-scality) [inline buildchain/buildchain/salt_tree.py:368]: ditto, why removing the 99?

File: docs/quickstart/bootstrap.rst [hunk 4]
```diff
@@ -11,8 +11,8 @@ Mount this ISO file at the specific following path:
 
 .. code-block:: shell
 
-   root@bootstrap $ mkdir -p /srv/scality/metalk8s-|version|
```
Comment (slaperche-scality) [inline docs/quickstart/bootstrap.rst:14]: Why are you hardcoding stuff? Now it will be wrong for 2.1 and 2.2…

File: salt/metalk8s/repo/installed.sls [hunk 5]
```diff
@@ -29,7 +29,7 @@ Generate repositories nginx configuration:
 Deploy container registry nginx configuration:
   file.managed:
     - name: {{ nginx_registry_path }}
-    - source: salt://{{ slspath }}/files/{{ nginx_registry }}
```
Comment (slaperche-scality) [inline salt/metalk8s/repo/installed.sls:32]: Why this change? Is `{{ nginx_registry }}` now unused?

**Review by slaperche-scality — CHANGES_REQUESTED** [review]

File: docs/quickstart/bootstrap.rst (hunk 4 above)
Comment (TeddyAndrieux) [reply docs/quickstart/bootstrap.rst:14]: Because `|version|` is not render in code-block that not working before this PR

File: salt/metalk8s/repo/installed.sls (hunk 5 above)
Comment (TeddyAndrieux) [reply salt/metalk8s/repo/installed.sls:32]: It's used to compute the path `nginx_registry_path`

File: buildchain/buildchain/salt_tree.py (hunk 2 above)
Comment (TeddyAndrieux) [reply buildchain/buildchain/salt_tree.py:280]: Just removed the prefix as it's not needed to have those prefix in our salt tree

File: buildchain/buildchain/salt_tree.py (hunk 3 above)
Comment (TeddyAndrieux) [reply buildchain/buildchain/salt_tree.py:368]: Same as above, prefix not needed and we don't need/want to have the version in the path as it's obvious that it's the file of our current version

File: salt/_modules/metalk8s.py (hunk 1 above)
Comment (TeddyAndrieux) [reply salt/_modules/metalk8s.py:228]: Not related directly but it's already in a separate commit

File: docs/quickstart/bootstrap.rst [hunk 6]
```diff
@@ -11,8 +11,8 @@ Mount this ISO file at the specific following path:
...
-   root@bootstrap $ mount <path-to-iso> /srv/scality/metalk8s-|version|
+   root@bootstrap $ mkdir -p /srv/scality/metalk8s-2.0.0-dev
```
Comment (charlesprost) [inline docs/quickstart/bootstrap.rst]: Are we sure this is going to work? Last time I've tried changing this version number it failed.

File: buildchain/buildchain/iso.py [hunk 7]
```diff
@@ -120,7 +120,7 @@ def task__iso_render_bootstrap() -> types.TaskDict:
     return helper.TemplateFile(
         source=constants.ROOT/'scripts'/'bootstrap.sh.in',
         destination=constants.ISO_ROOT/'bootstrap.sh',
-        context={'VERSION': constants.SHORT_VERSION},
+        context={'VERSION': constants.VERSION},
         file_dep=[constants.VERSION_FILE],
         task_dep=['_iso_mkdir_root'],
     ).task
```
Comment (charlesprost) [inline buildchain/buildchain/iso.py:126]: Why not remove SHORT_VERSION from below?

File: buildchain/buildchain/salt_tree.py (hunk 2 above)
Comment (slaperche-scality) [reply buildchain/buildchain/salt_tree.py:280]: Not related to the VERSION switch then, so should be in a dedicated commit (where you could explain it's unneeded in the commit message).

File: salt/_modules/metalk8s.py (hunk 1 above)
Comment (slaperche-scality) [reply salt/_modules/metalk8s.py:228]: yes, and it's a good point.

Still, this seems to fix a bug ("to not block boot in case an iso no longer here") so I think it deserves its own issues/PR.

File: docs/quickstart/bootstrap.rst (hunk 4 above)
Comment (slaperche-scality) [reply docs/quickstart/bootstrap.rst:14]: Then we should probably find an alternative.

Comment (gdemonet) [reply docs/quickstart/bootstrap.rst:14]: Replace the `code-block` directive with `parsed-literal`, should work (though lost highlighting).

File: salt/_modules/metalk8s.py (hunk 1 above)
Comment (TeddyAndrieux) [reply salt/_modules/metalk8s.py:228]: It's already refer to the issue #924

File: docs/quickstart/bootstrap.rst (hunk 4 above)
Comment (TeddyAndrieux) [reply docs/quickstart/bootstrap.rst:14]: Nice, I will do that thanks

File: buildchain/buildchain/iso.py (hunk 7 above)
Comment (TeddyAndrieux) [reply buildchain/buildchain/iso.py:126]: Because I think we may need this one day,
IMO we can keep it

File: salt/_modules/metalk8s.py (hunk 1 above)
Comment (slaperche-scality) [reply salt/_modules/metalk8s.py:228]: Ok, good enough then

File: buildchain/buildchain/salt_tree.py (hunk 2 above)
Comment (TeddyAndrieux) [reply buildchain/buildchain/salt_tree.py:280]: Done

**Comment by TeddyAndrieux** [comment]
@bert-e reset

**Review by slaperche-scality — APPROVED** [review]

File: docs/quickstart/bootstrap.rst (hunk 6 above)
Comment (TeddyAndrieux) [reply docs/quickstart/bootstrap.rst]: What do you mean by "it failed" ?

**Comment by TeddyAndrieux** [comment]
@bert-e approve
